### PR TITLE
Signer new circuits mantapay v1

### PR DESCRIPTION
**Pull request by Apokalip**
Changes for redesign and mantapay-v1 with the new circuit
---

Before we can merge this PR, please make sure that all the following items have been checked off:

- [ ] Linked to an issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Added **one** line describing your change in [`CHANGELOG.md`](https://github.com/manta-network/manta-signer/blob/main/CHANGELOG.md) and added the appropriate `changelog` label to the PR.
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.

### CODE REVIEW

**Review by zqhxuyuan — DISMISSED** [review]

File: ui/src-tauri/src/main.rs [hunk 1]
```diff
@@ -518,15 +520,16 @@ async fn reset_account(
...
 /// Returns receiving keys to front end to display once user is logged in.
 #[tauri::command]
-async fn receiving_keys(server_store: State<'_, ServerStore>) -> Result<Vec<String>, ()> {
+async fn address(server_store: State<'_, ServerStore>) -> Result<String, ()> {
     if let Some(store) = &mut *server_store.lock().await {
-        let keys = store
-            .get_receiving_keys(Message {
+        let key = store
+            .get_address(Message {
+                // TODO: do we need to passing network?
```
Comment (AlexParshh) [inline ui/src-tauri/src/main.rs:527]: @zqhxuyuan I don't think the network matters because the address is the same for all

**Review by AlexParshh — DISMISSED** [review]

File: ui/src-tauri/src/main.rs (hunk 1 above)
Comment (Apokalip) [reply ui/src-tauri/src/main.rs:527]: It doesnt hurt us in any way honestly, keeps it readable and will probably be easier to know where to change in the future if we have a need to pass different networks

**Review by Kevingislason — APPROVED** [review]

File: ui/src-tauri/src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Apokalip) [reply ui/src-tauri/src/main.rs:527]: No We left it just to make it easier in the future for readability if we actually need to control the Network argument

**Review by AlexParshh — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -43,13 +43,14 @@ derivative = { version = "2.2.0", default-features = false, features = ["use_cor
...
-manta-crypto = { git = "https://github.com/manta-network/manta-rs", tag = "v0.5.7", default-features = false, features = ["getrandom"] }
-manta-parameters = { git = "https://github.com/manta-network/manta-rs", tag = "v0.5.7", default-features = false, features = ["download"] }
-manta-pay = { git = "https://github.com/manta-network/manta-rs", tag = "v0.5.7", default-features = false, features = ["bs58", "groth16", "network", "serde", "wallet"] }
-manta-util = { git = "https://github.com/manta-network/manta-rs", tag = "v0.5.7", default-features = false }
+manta-accounting = { git = "https://github.com/manta-network/manta-rs", default-features = false, features = ["cocoon-fs"] }
+manta-crypto = { git = "https://github.com/manta-network/manta-rs", default-features = false, features = ["getrandom"] }
+manta-parameters = { git = "https://github.com/manta-network/manta-rs", default-features = false, features = ["download"] }
+manta-pay = { git = "https://github.com/manta-network/manta-rs", default-features = false, features = ["bs58", "groth16", "serde", "wallet", "network", "parameters"] }
+manta-util = { git = "https://github.com/manta-network/manta-rs", default-features = false }
```
Comment (Kevingislason) [inline Cargo.toml:50]: we intend to get rid of the tag here, right?

**Review by Kevingislason — APPROVED** [review]

File: Cargo.toml (hunk 2 above)
Comment (Apokalip) [reply Cargo.toml:50]: No, its just that the display change in manta-rs is in main but there is currently no release with that change, ill update this when there is 0.5.9 which should contain that part

**Review by AlexParshh — APPROVED** [review]

**Comment by Apokalip** [comment]
Unsigned commits preventing merge, copied to #228

**Pull request closed without merging**